Source/Ansible/ansible-inventory.env

```diff
@@ -0,0 +1,98 @@
+_t4dSrcAnsibleInventoryAdd(){
+# Generated From $Tools4Dev_PATH/Engine/G4d/templates.env
+# v1.8.0
+###### _t4dSrcAnsibleInventoryAdd
+#   - Name:
+#       _t4dSrcAnsibleInventoryAdd
+###
+#   - Synopsis:
+#       add host in Ansible Inventory
+###
+#   - Usage:
+#        _t4dSrcAnsibleInventoryAdd <file> <group> <host>
+###
+#   - Note:
+#       <Required>
+#       [Optionnal]
+#       {XXXXXXX}       = Value To Be Changed, if {URL} then replace {URL} with a valid URL
+###
+###### DOCUMENTATION BLOC #####
+########### CODE BLOC #########
+    local _t4dErrorCode=1
+    local _File="$1"
+    local _Group="$2"
+    local _Host="$3"
+
+    _t4dSrcYamlOpenFile "$_File"
+    _t4dSrcYamlUpdateLine "$T4D_PROJECT_LOG_PATH/yaml/_t4dSrcYamlParseFile-$_File.in" ">groups[\"$_Group\"]" -type "LIST" -indent 0 -key "groups" -data "\"$_Group\""
+    _t4dSrcYamlUpdateLine "$T4D_PROJECT_LOG_PATH/yaml/_t4dSrcYamlParseFile-$_File.in" ">groups[\"$_Group\"]>- $_Host"
+
+    _t4dSrcYamlCloseFile "$_File" >/dev/null
+
+    cat "$_File" | sed 's|^  - ||g' | sed 's|- groups: "|\n\[|g' | sed 's|"|\]|g' > "$_File"
+
+###############################
+}
+
+
+_t4dSrcAnsibleInventoryFind(){
+# Generated From $Tools4Dev_PATH/Engine/G4d/templates.env
+# v1.8.0
+###### _t4dSrcAnsibleInventoryFind
+#   - Name:
+#       _t4dSrcAnsibleInventoryFind
+###
+#   - Synopsis:
+#       find in inventory
+###
+#   - Usage:
+#        _t4dSrcAnsibleInventoryFind <file> <host>
+###
+#   - Note:
+#       <Required>
+#       [Optionnal]
+#       {XXXXXXX}       = Value To Be Changed, if {URL} then replace {URL} with a valid URL
+###
+###### DOCUMENTATION BLOC #####
+########### CODE BLOC #########
+    local _t4dErrorCode=1
+    local _File="$1"
+    local _Host="$2"
+
+    _t4dSrcYamlOpenFile "$_File" >/dev/null
+
+    local _res=$(_t4dSrcYamlSearch "$T4D_PROJECT_LOG_PATH/yaml/_t4dSrcYamlParseFile-$_File.yml.in" "$_Host") 
+    echo $_res > "$T4D_PROJECT_LOG_PATH/yaml/_t4dSrcYamlParseFile-$_File.yml.in.tmp"
+    
+    _t4dSrcYamlCloseFile "$_File.yml" "$T4D_PROJECT_LOG_PATH/yaml/_t4dSrcYamlParseFile-$_File.yml.in.tmp"
+
+    rm $_File.yml
+    rm "$T4D_PROJECT_LOG_PATH/yaml/_t4dSrcYamlParseFile-$_File.yml.in"
+
+###############################
+}
+
+_t4dSrcAnsibleSortGroups(){
+# Generated From $Tools4Dev_PATH/Engine/G4d/templates.env
+# v1.8.0
+###### _t4dSrcAnsibleSortGroups
+#   - Name:
+#       _t4dSrcAnsibleSortGroups
+###
+#   - Synopsis:
+#       sort groups in inventory serialise file
+###
+#   - Usage:
+#        _t4dSrcAnsibleInventoryFind <file>
+###
+#   - Note:
+#       <Required>
+#       [Optionnal]
+#       {XXXXXXX}       = Value To Be Changed, if {URL} then replace {URL} with a valid URL
+###
+###### DOCUMENTATION BLOC #####
+########### CODE BLOC #########
+    cat "$1" | cut -d '|' -f1 | rev | grep -Eo '>.*' | rev | sort | uniq | sed 's|>$||g'
```

Comment:
you did not use `chf` to create this. 

Please add 
`local _t4dErrorCode=1`
and `local _file=$1`

and do not cat if file does not exist. 

`if [[ -e $_file ]] `